src/helpers/formatNumberWithSpaces.ts

```diff
@@ -0,0 +1,3 @@
+export function formatNumberWithSpaces(number: number) {
+  return number.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ' ')
```

Comment:
zastanawiam się - a co jeśli chcielibyśmy formatować czasem stringa? ;p To wtedy robimy na zewnątrz funkcji konwersję string do number i potem w tej funkcji number do string. Może lepiej zrobić, żeby ta funkcja przyjmowała stringa jako argument, wtedy unikniemy tutaj konwersji i zrobimy ją na zewnątrz. I wtedy jeśli będziemy chcieli kiedyś formatować stringa tą funkcją to nie będzie tej podwójnej konwersji 🤔

Reply:
Według mnie spowoduje to troche brak czytelności, funkcja dostanie argument formatNumberWithSpaces(number: string), co jest trochę mylące. 
Dodatkowo trzeba zastanowić się, ile sytuacji będzie takich, że z backend'u jakiegoś number'a dostaniemy jako string, bo jeśli jakiś znikomy procent, to według mnie lepiej zostać przy tym obecnym rozwiazaniu, nie będziemy musieli w kodzie robić tej konwersji (number na string), tylko przekażemy number do funkcji i zrobi się samo, bez pamiętania o tym.
Natomiast jeśli będzie dużo sytuacji, gdzie będziemy chcieli tym formatować stringa, to masz rację.

Reply:
No wiesz, bo at the end of the day formatujesz stringa i wynikowo dostajesz stringa, jakby coś do wyświetlenia. Albo np. jak masz input w który w teorii możesz wpisać tylko liczby numeryczne to jak weźmiesz wartość tego pola to ono będzie stringiem (no chyba, że ustawisz type number ale wtedy już wiesz, że są problemy, bo ostatnio mieliśmy XD) No są przypadki gdzie być może trzeba będzie formatowac stringa i wtedy ta funkcja jest mniej uniwersalna bo są te podwójne konwersje. Ale to jest kwestia gustu tak serio i zdecydowania się na jedną opcje. Zostawmy jak jest i zobaczmy jaka będzie potrzeba ^^

Reply:
Po obszernej dyskusji zostawiamy i zobaczymy co będzie :)